imod/mf6/simulation.py

```diff
@@ -941,12 +941,22 @@ def split(self, submodel_labels: xr.DataArray) -> Modflow6Simulation:
                 new_simulation[solution_name].add_model_to_solution(new_model_name)
 
         exchanges = []
-        for model_name, model in original_models.items():
-            if isinstance(model, GroundwaterFlowModel):
-                exchanges += exchange_creator.create_gwfgwf_exchanges(
-                    model_name, model.domain.layer
+        flow_models = self.get_models_of_type("gwf6")
+        if len(flow_models) != 1:
+            raise ValueError(
+                "splitting of simulations with more (or less) than 1 flow model currently not supported"
+            )
+        flow_model_name = list(flow_models.keys())[0]
+        flow_model = self[flow_model_name]
```

Comment:
Dont know if this is better but you can combine these 2 lines into:
`flow_model_name, flow_model= next(iter(gwf_models.items()))`

Reply:
that line is gone.  A test failed in which mulltiple flow models were used in the same simulation, and hence we loop now over the flow models, rather thatn focussing on the first one.